src/overlays/actors/ovl_Obj_Um/z_obj_um.c

```diff
@@ -46,131 +80,1951 @@ const ActorInit Obj_Um_InitVars = {
+    ObjUm_RotatePlayer(this, play, 0x7FFF);
+    this->wheelRot += 2000;
+    this->flags |= OBJ_UM_FLAG_0010;
+    this->flags |= OBJ_UM_FLAG_PLAYING_MINIGAME;
+    ObjUm_UpdateAnim(this, play, 1);
+
+    if (ObjUm_UpdatePath(this, play) == OBJUM_PATH_STATE_3 && this->unk_4DC == 0) {
+        this->unk_4DC = 1;
+    } else if (this->unk_4DC > 0) {
+        if (this->unk_4DC == 1) {
+            s32 i;
+
+            this->arePotsBroken = true;
+
+            for (i = 0; i < ARRAY_COUNT(this->potsLife); i++) {
+                if (this->potsLife[i] != 1) {
+                    this->arePotsBroken = false;
+                    break;
+                }
+            }
+
+            this->flags |= OBJ_UM_FLAG_MINIGAME_FINISHED;
+
+        dummy_label:;
+        }
+
+        this->unk_4DC++;
+    }
+
+    if (this->flags & OBJ_UM_FLAG_MINIGAME_FINISHED) {
+        s32 sp20 = ActorCutscene_GetAdditionalCutscene(this->dyna.actor.cutscene);
+
+        if (this->arePotsBroken) {
+            sp20 = ActorCutscene_GetAdditionalCutscene(sp20);
+        }
+        if (ActorCutscene_GetCanPlayNext(sp20)) {
+            ActorCutscene_StartAndSetUnkLinkFields(sp20, &this->dyna.actor);
+            ObjUm_SetupAction(this, ObjUm_RunMinigame);
+            this->flags &= ~OBJ_UM_FLAG_PLAYING_MINIGAME;
+        } else {
+            ActorCutscene_SetIntentToPlay(sp20);
+        }
+    }
+
+    Actor_MoveWithGravity(&this->dyna.actor);
+    func_80B78DF0(this, play);
+}
+
+void func_80B7A7AC(ObjUm* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    this->unk_4DC = 0;
+    this->arePotsBroken = false;
+    player->stateFlags1 &= ~0x20;
+    ObjUm_SetPlayerPosition(this, play);
+    ObjUm_RotatePlayer(this, play, 0x7FFF);
+    func_80B78EBC(this, play);
+    this->flags |= OBJ_UM_FLAG_0010;
+    func_80B78DF0(this, play);
+    this->flags |= 4;
+    ObjUm_UpdateAnim(this, play, 1);
+    ObjUm_SetupAction(this, func_80B7A614);
+}
+
+void func_80B7A860(ObjUm* this, PlayState* play) {
+    ObjUm_SetPlayerPosition(this, play);
+    ObjUm_RotatePlayer(this, play, 0);
+    this->flags |= OBJ_UM_FLAG_0004;
+
+    if (play->csCtx.frames == 449) {
+        ObjUm_InitBandits(this, play);
+    } else if (play->csCtx.frames >= 450) {
+        func_80B78DF0(this, play);
+    }
+
+    if (play->csCtx.state == 0) {
+        ActorCutscene_Stop(this->dyna.actor.cutscene);
+        ObjUm_SetupAction(this, func_80B7A7AC);
+    }
+
+    switch (play->msgCtx.currentTextId) {
+        case 0x33BA:
+            this->unk_4CC = 2;
+            this->mouthTexIndex = 3;
+            break;
+
+        // "We'll go through here as fast as we can"
+        case 0x33BB:
+            if (IS_ZERO(this->skelAnime.curFrame) && !(this->flags & OBJ_UM_FLAG_1000)) {
+                this->flags |= OBJ_UM_FLAG_1000;
+                this->unk_4CC = 4;
+                this->mouthTexIndex = 0;
+            } else if (IS_ZERO(this->skelAnime.morphWeight)) {
```

Comment:
From what I understand, the `IS_ZERO` macro is mainly for collision and shouldn't be used outside that system, since "zero" checks are completely inconsistent in actors. So I would not macro-ize this (applies to all `IS_ZERO`'s in this file)

Reply:
I don't think this is necessarily true, basically nothing checks this condition in OoT: literally 3 actors.

Reply:
(And currently it's the same 3 in MM, this is number 4.)